core/src/Kokkos_Array.hpp

```diff
@@ -356,6 +356,32 @@ struct KOKKOS_DEPRECATED
 template <typename T, typename... Us>
 Array(T, Us...)->Array<T, 1 + sizeof...(Us)>;
 
+namespace Impl {
+
+template <typename T, size_t N, size_t... I>
+KOKKOS_FUNCTION constexpr Array<std::remove_cv_t<T>, N> to_Array_impl(
+    T (&a)[N], std::index_sequence<I...>) {
+  return {{a[I]...}};
+}
+
+template <typename T, size_t N, size_t... I>
+KOKKOS_FUNCTION constexpr Array<std::remove_cv_t<T>, N> to_Array_impl(
+    T(&&a)[N], std::index_sequence<I...>) {
+  return {{std::move(a[I])...}};
+}
+
+}  // namespace Impl
+
+template <typename T, size_t N>
+KOKKOS_INLINE_FUNCTION constexpr auto to_Array(T (&a)[N]) {
```

Comment:
I would drop the "inline" here and below.

Reply:
Fixed (or will be the next time I push the repo up).